File: src/main/java/ro/nicolaemariusghergu/queryit/mapper/CategoryMapper.java
package ro.nicolaemariusghergu.queryit.mapper;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;
import ro.nicolaemariusghergu.queryit.dto.CategoryDto;
import ro.nicolaemariusghergu.queryit.model.Category;

@Mapper
public interface CategoryMapper {

    CategoryMapper INSTANCE = Mappers.getMapper(CategoryMapper.class);

    CategoryDto categoryToCategoryDto(Category category);

    Category categoryDtoToCategory(CategoryDto categoryDto);

}

File: src/main/kotlin/ro/nicolaemariusghergu/queryit/mapper/CategoryMapper.kt
package ro.nicolaemariusghergu.queryit.mapper

import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers
import ro.nicolaemariusghergu.queryit.dto.CategoryDto
import ro.nicolaemariusghergu.queryit.model.Category

@Mapper
interface CategoryMapper {
    fun categoryToCategoryDto(category: Category): CategoryDto
    fun categoryDtoToCategory(categoryDto: CategoryDto): Category

    companion object {
        val INSTANCE: CategoryMapper = Mappers.getMapper(CategoryMapper::class.java)
    }
}